### PR TITLE
Add magefile.go for compiling proto

### PR DESCRIPTION
**Pull request by severinson**

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+	case "darwin/amd64":
+		return "osx-x86_64", nil
+	case "darwin/arm64":
+		return "osx-aarch_64", nil
+	case "linux/386":
+		return "linux-x86_32", nil
+	case "linux/amd64":
+		return "linux-x86_64", nil
+	case "linux/arm64":
+		return "linux-aarch_64", nil
```
Comment (dejanzele) [inline magefile.go:330]: should it be `linux-aarch_64` (double a)

File: magefile.go [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+	case "windows/386":
+		return "win32", nil
+	case "windows/amd64":
+		return "win64", nil
+	}
+	return "", fmt.Errorf("protoc not supported on %s/%s", runtime.GOOS, runtime.GOARCH)
```
Comment (dejanzele) [inline magefile.go:336]: I'd rather not use `fmt.Errorf` as it is deprecated, `github.com/pkg/errors` is better but not maintained.
I think https://github.com/go-faster/errors is the successor.

File: magefile.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+	}
+	return sh.OutCmd("protoc"), nil
+}
+
+func exists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
```
Comment (dejanzele) [inline magefile.go:313]: can you replace this block with just `return false, err`, caller anyways need to check is `err != nil` and you return false in both cases.

File: magefile.go [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.ContentLength < 1000 {
+		contents, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("failed to download protoc: unexpected response '%s'", string(contents))
```
Comment (dejanzele) [inline magefile.go:230]: same for `fmt.Errorf`

File: magefile.go (hunk 1 above)
Comment (severinson) [reply magefile.go:330]: There's already a double a?

File: magefile.go (hunk 2 above)
Comment (severinson) [reply magefile.go]: Let's use `github.com/pkg/errors` since we use that everywhere else.

File: magefile.go (hunk 3 above)
Comment (severinson) [reply magefile.go]: Done

File: magefile.go (hunk 4 above)
Comment (severinson) [reply magefile.go]: Done

File: magefile.go (hunk 2 above)
Comment (severinson) [reply magefile.go]: Done

File: .github/dependabot.yml [hunk 5]
```diff
@@ -1,49 +1,49 @@
-# Basic dependabot.yml file with
```
Comment (severinson) [inline .github/dependabot.yml:1]: Fixed line endings

File: .github/workflows/codeql-analysis.yml [hunk 6]
```diff
@@ -1,77 +1,77 @@
-# For most projects, this workflow file will not need changing; you simply need
```
Comment (severinson) [inline .github/workflows/codeql-analysis.yml:1]: Fixed line endings

File: .gitpod.Dockerfile [hunk 7]
```diff
@@ -1,5 +1,5 @@
-FROM golang:1.18.5-buster
```
Comment (severinson) [inline .gitpod.Dockerfile:1]: Fixed line endings

File: .gitpod.yml [hunk 8]
```diff
@@ -1,16 +1,16 @@
-image:
```
Comment (severinson) [inline .gitpod.yml:1]: Fixed line endings

File: client/DotNet/ArmadaProject.Io.Client/ArmadaProject.Io.Client.csproj [hunk 9]
```diff
@@ -1,28 +1,28 @@
-<Project Sdk="Microsoft.NET.Sdk">
```
Comment (severinson) [inline client/DotNet/ArmadaProject.Io.Client/ArmadaProject.Io.Client.csproj:1]: Fixed line endings

**Comment by severinson** [comment]
E2e tests failed due to image pull limits. This shouldn't have any impact on e2e tests.

**Review by dejanzele — APPROVED** [review]

**Pull request merged**